### PR TITLE
Multi Istio control planes support

### PR DESCRIPTION
**Pull request by waynz0r**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| API breaks?     | no
| Deprecations?   | no
| Related tickets | 
| License         | Apache 2.0

### What's in this PR?

This PR adds multi control plane support, so users can have multiple Istio control planes in a cluster, even within a single namespace.

### Why?

Having the ability to support multiple control planes gives option to support canary style upgrade for the user and allowing gradual upgrade process instead of the in-place one.

### Checklist

- [x] Implementation tested

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -11,6 +11,7 @@ GOLANGCI_VERSION = 1.23.8
 LICENSEI_VERSION = 0.1.0
 KUBEBUILDER_VERSION = 1.0.8
 KUSTOMIZE_VERSION = 2.0.3
+ISTIO_VERSION = 1.6.5
```
Comment (Laci21) [inline Makefile:14]:
```suggestion
ISTIO_VERSION = 1.6.7
```

File: pkg/apis/istio/v1beta1/istio_types.go [hunk 2]
```diff
@@ -29,7 +30,14 @@ import (
 	"github.com/banzaicloud/istio-operator/pkg/util"
 )
 
-const supportedIstioMinorVersionRegex = "^1.6"
+const (
+	supportedIstioMinorVersionRegex = "^1.6"
+)
+
+var (
+	SupportedIstioVersion = "1.6.3"
```
Comment (Laci21) [inline pkg/apis/istio/v1beta1/istio_types.go:38]:
```suggestion
	SupportedIstioVersion = "1.6.7"
```

File: pkg/apis/istio/v1beta1/istio_types.go [hunk 3]
```diff
@@ -29,7 +30,14 @@ import (
...
+	Version               = "0.6.4"
```
Comment (Laci21) [inline pkg/apis/istio/v1beta1/istio_types.go:39]:
```suggestion
	Version               = "0.6.7"
```

File: pkg/apis/istio/v1beta1/istio_types.go [hunk 4]
```diff
@@ -662,7 +673,7 @@ type HTTPProxyEnvs struct {
 // IstioSpec defines the desired state of Istio
 type IstioSpec struct {
 	// Contains the intended Istio version
-	// +kubebuilder:validation:Pattern=^1.6
+	// +kubebuilder:validation:Pattern=^1.
```
Comment (Laci21) [inline pkg/apis/istio/v1beta1/istio_types.go:676]: Is this change intended?

Comment (waynz0r) [reply pkg/apis/istio/v1beta1/istio_types.go:676]: It is, the crd should allow 1.6 and 1.7 as well.
